library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -652,11 +639,7 @@ def filter_interfacetype(devmap)
     def ConcealSecrets1(ifcfg)
       ifcfg = deep_copy(ifcfg)
       return nil if ifcfg.nil?
-      out = Builtins.mapmap(ifcfg) do |k, v|
-        v = "CONCEALED" if Builtins.contains(@SensitiveFields, k) && v != ""
-        { k => v }
-      end
-      deep_copy(out)
+      ifcfg.each { |k, v| ifcfg[k] = "CONCEALED" if @SensitiveFields.include?(k) && v != "" }
```

Comment:
You could split this part in:

1) finding what `ifcfg` elements should be `concealed`
2) setting `concealed` value

Anyway, if you want to keep it as it is, I would avoid the "if" at the end.

Just a suggestion.